### DISCUSSION TITLE
Backup UI

### DISCUSSION THREAD

**Discussion by marcoczen** [cockpit-project/cockpit#16302] [Q&A] [answered]
Hi,

How do you guys do backups of your VM ?
Will cockpit have a backup module down the road ? 

Thanks.

**Comment by marcoczen** [comment]
I am trying to remove the label 'enhancement' but cant ...I would prefer the label 'Discussion'
:-)

**Comment by marusak** [comment]
> How do you guys do backups of your VM ?
> Will cockpit have a backup module down the road ?

Cockpit has support for snapshoting. Is that you are asking about? https://cockpit-project.org/blog/cockpit-225.html

> I am trying to remove the label 'enhancement' but cant ...I would prefer the label 'Discussion'

It is fine :) Notice, that there is also 'Discussions' for general questions.

**Comment by marcoczen** [comment]
Hi ..Is 'Backup' the same as 'snapshotting' ?  I am too new to the KVM ecosystem to understand nuances if they do exist between these two terms.

I am thinking of 'Backup' in the regular sense, where if i needed i can do a 'Restore' and the VM is operational and accesible again?
Ideally i should be able to keep the backup in a NFS share or some other host ?

Something like the Proxmox Backup Server ?

**Reply by martinpitt** [reply]
"Snapshot" means that the current file system contents  of the qcow2 VM file gets "remembered" and labelled; on restoring, it then gets reset to the previous state, so that any changes since the snapshot get reverted.

It has nothing to do with generic file backup where you copy a bunch of files/directories to some other place -- but that is not at all specific to VM management, you should just include the VM storage paths (/var/lib/libvirt/...) in your backup program.

So I misunderstood your question. There is no current Cockpit UI for controlling backup programs. There might be some day (possibly/hopefully also from some third party), but it's not currently planned.

**Reply by marcoczen** [reply]
Hi ..can we export the snapshot ? Is it a single file ? And then restore it in another host ( that is also running cockpit ) for example ?

**Reply by martinpitt** [reply]
At the moment, snapshots are stored inside the .qcow2 file. So in some sense you *can* restore it on another host by copying the whole VM. (This is independent of cockpit)

**Reply by marcoczen** [reply]
So copying out the qcow2 will allow me to re-import the VM  into another cockpit installation ? Nice...
Any caveats ?

**Comment by marcoczen** [comment]
Ok noted. I understand that the snapshot is stored inside the qcow2 file. 
And we can copy out the  the whole .qcow2  to another host and recreate the VM there.

DO we need to stop the VM before we copy out the .qcow2 file ?

**Reply by martinpitt** [reply]
The Machines page's builtin migration feature can copy running VMs. But normal backup software or simple file copy  is not aware of these files changing on the fly during copy, so VMs should better be shut down, yes.

**Comment by marcoczen** [comment]
Noted. Thanks. 

Final clarification....Just wondering why something like 'Proxmox Backup Server' is not being considered for  backing up cockpit VMs? 
Not generic backups but specifically VMs ?

Or do i have to make a request to the cockpit-machines team  ( https://github.com/cockpit-project/cockpit-machines ) ?

**Reply by martinpitt** [reply]
It's not like we'd say "no" to a backup UI (in fact, I think it'd be great to have one!) -- it's just like there is none right now, and difficult as backups are, there is currently no design idea/proposal what it should do, nor someone who has the time and desire to implement it :grin:

**Comment by marcoczen** [comment, marked as answer]
Busy ?  No problem. Many thanks.
Keep up the great work.

Closed.

**Comment by mmljmb3** [comment]
Has there been any new thoughts on this? It has been years.
